main/src/ca/uwaterloo/flix/tools/pkg/FlixPackageManager.scala

```diff
@@ -90,6 +105,90 @@ object FlixPackageManager {
     }
   }
 
+  //TODO: test!
+  /**
+    * Finds all transitive dependencies for a list of Flix dependencies,
+    * and puts the list in `flixSet`. Also finds any Maven dependencies,
+    * and puts those in `mavenSet`. Returns an error if anything goes wrong.
+    */
+  private def findTransitiveDependencies(flixDepsManifest: List[FlixDependency], p: Path): Result[Unit, PackageError] = {
+    workSet = collection.mutable.Set.empty[FlixDependency]
+
+    flixDepsManifest.foreach(dep => {workSet.addOne(dep); flixSet.addOne(dep)})
+    val tempDir = Files.createTempDirectory(p, "")
+
+    while(workSet.nonEmpty) {
+      val dep = workSet.head
+      workSet.remove(dep)
+
+      for(
+        proj <- GitHub.parseProject(s"${dep.username}/${dep.projectName}");
+        release <- GitHub.getSpecificRelease(proj, dep.version)
+      ) yield {
+        val tomlFiles = release.assets.filter(_.name.endsWith(".toml"))
+        traverse(tomlFiles)(tomlFile => {
+          val assetName = tomlFile.name
+          val tomlPath = tempDir.resolve(s"${dep.username}-${dep.projectName}-$assetName")
+          processTomlFile(tomlFile, tomlPath)
+        }) match {
+          case Ok(_) =>
+          case Err(e) => return Err(e)
+        }
+      }
+    }
+
+    //TODO: should we save the toml files?
+    FileUtils.deleteDirectory(tempDir.toFile)
```

Comment:
Currently I download the toml files into a temporary directory and delete them when the exploration of transitive dependencies is over. Should I instead save the toml files?

Reply:
Lets same them, we should not deleted them. Nor do we have to re-download them.

Reply:
I would put them right next to the package files.